### DISCUSSION TITLE
How to use/verify MyTeslaMate Streaming?

### DISCUSSION THREAD

**Discussion by bjun626** [teslamate-org/teslamate#4161] [Q&A] [answered]
After changing to the Tesla's official api using `api.myteslamate.com` & `wss://streaming.myteslamate.com` I can't get an update from the Fleet Telemetry while driving. When looking at the MQTT data, it doesn't seem to be updating every 1 minute.

## Help Needed

Based on the results from the steps below, I presume the Tesla side has been sending the data to `telemetry.myteslamate.com` correctly, right now I'm unable to confirm teslamate is obtaining data from `wss://streaming.myteslamate.com`. 

- Is there a way to troubleshoot/analyse `wss://streaming.myteslamate.com`? 
- Are there any settings that I missed?

### 1. Changing Settings

In https://teslamate.example.com/settings, when I turned on **Streaming API** I can't get any updates while driving, and if I turned it off I will get an update but it seems like it's using the `- POLLING_DRIVING_INTERVAL=300`

**Streaming API** Seems like its not getting data from `wss://streaming.myteslamate.com`

### 2. Recheck Configuration

As per the docs, I've "*Pair your vehicle(s)*" and currently *Fleet Status* is **Synced** and added the below ENV
```
- TESLA_API_HOST=https://api.myteslamate.com
- TESLA_AUTH_HOST=https://api.myteslamate.com
- TESLA_AUTH_PATH=/api/oauth2/v3
- TESLA_WSS_HOST=wss://streaming.myteslamate.com
- TESLA_WSS_TLS_ACCEPT_INVALID_CERTS=true
- TESLA_WSS_USE_VIN=true
- TOKEN=?token=<REDACTED>
- POLLING_ASLEEP_INTERVAL=300
- POLLING_CHARGING_INTERVAL=450
- POLLING_DRIVING_INTERVAL=300
- POLLING_ONLINE_INTERVAL=450
- POLLING_DEFAULT_INTERVAL=120
- POLLING_MINIMUM_INTERVAL=120
```

### 3. Check Tesla's Fleet Config

Calling `https://api.myteslamate.com/api/1/vehicles/<CAR VIN>/fleet_telemetry_config?token=<REDACTED>` seems to be returning the correct configuration.

```
{
  "response": {
    "synced": true,
    "config": {
      "hostname": "telemetry.myteslamate.com",
      "ca": "-----BEGIN CERTIFICATE-----<telemetry.myteslamate.com cert>-----END CERTIFICATE-----",
      "exp": 1755696260,
      "port": 443,
      "fields": {
        "ACChargingPower": {
          "interval_seconds": 60
        },
        "DCChargingPower": {
          "interval_seconds": 60
        },
        "EstBatteryRange": {
          "interval_seconds": 60
        },
        "Gear": {
          "interval_seconds": 60
        },
        "GpsHeading": {
          "interval_seconds": 60
        },
        "Location": {
          "interval_seconds": 60
        },
        "Odometer": {
          "interval_seconds": 60
        },
        "RatedRange": {
          "interval_seconds": 60
        },
        "Soc": {
          "interval_seconds": 60
        },
        "VehicleSpeed": {
          "interval_seconds": 60
        }
      },
      "alert_types": [
        "service"
      ]
    }
  }
}
```

### When Driving

During driving, I can see that the **Location Sharing** icon is constantly on as compared to polling which will only pop up once a few minutes.

![Screenshot 2024-08-27 142759](https://github.com/user-attachments/assets/da152e83-3e63-4896-aa6e-16d261140c70)

## Vehicle Details

- Model 3
- 2024.26.7

## Teslamate Details

Was using `1.30.1`, updated to `1.30.2-dev / edge`. Still can't get the data.

**Comment by jlestel** [comment, marked as answer]
Tesla Fleet streaming had evolved in August leading to a stop of the streaming forward: https://github.com/teslamotors/fleet-telemetry/issues/210

The forwarder was updated in early September: https://github.com/MyTeslaMate/websocket

Ps: questions related to the MyTeslamate API should rather be asked on the myteslamate.com chat.
